.github/workflows/synthetics-stack-e2e.yml

```diff
@@ -0,0 +1,76 @@
+#
+# This workflow is triggered by a comment on a PR and runs the synthetics stack e2e tests
+# It is responsible for setting the status check `synthetics-stack-e2e` to success or failure,
+# which was set to failure by the synthetics-stack-e2e-init.yml workflow.
+#
+---
+name: synthetics-check-run
+
+on:
+  issue_comment:
+    types:
+      - created
+
+jobs:
+  run-synthetics-stack-e2e:
+    if: ${{ github.event.issue.pull_request && startsWith(github.event.comment.body, '/synthetics-stack-e2e') }}
+    runs-on: ubuntu-latest
+    env:
+      RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}?pr=${{ github.event.issue.number }}
+    permissions:
+      statuses: write # Needed to set the commit status
+      pull-requests: write # Needed to react to the comment
+    steps:
+      - name: Get PR head SHA
```

Comment:
I wonder whether `elastic/apm-pipeline-library/.github/actions/run-synthetics-stack-e2e@current` could contains this whole GitHub action. This could help us with reducing the complexity in this repository. 

```
...
jobs:
  run-synthetics-stack-e2e:
    if: ${{ github.event.issue.pull_request && startsWith(github.event.comment.body, '/synthetics-stack-e2e') }}
    runs-on: ubuntu-latest
    permissions:
      statuses: write # Needed to set the commit status
      pull-requests: write # Needed to react to the comment
    steps:
      - uses: elastic/apm-pipeline-library/.github/actions/run-synthetics-stack-e2e@current
        with:
          vault-url: ${{ secrets.OBLT_VAULT_ADDR }}
          vault-role-id: ${{ secrets.OBLT_VAULT_ROLE_ID }}
          vault-secret-id: ${{ secrets.OBLT_VAULT_SECRET_ID }}
```

Reply:
My previous implementation it actually was like that.

In my thinking I put it here, because the the other pipeline creates the initial failing commit status.

Hence, I thought it would be wiser to keep that logic also here.

I guess, to keep us both happy.. I will also create an action for the second workflow.

Reply:
> In my thinking I put it here, because the the other pipeline creates the initial failing commit status.

Can you converge both workflows with two different jobs? One for each event?

```
on:
  pull_request_target:
    paths:
      - "x-pack/plugins/synthetics/**"
  issue_comment:
    types:
      - created
env:
  CHECK_NAME: your-check-name
jobs:
  run-synthetics-stack-e2e:
    if: ${{ github.event.issue.pull_request && startsWith(github.event.comment.body, '/synthetics-stack-e2e') }}
   ....
      - uses: elastic/apm-pipeline-library/.github/actions/run-synthetics-stack-e2e@current
        with:
           ...
           check-name: ${{ env.CHECK_NAME }}

  init-synthetics-stack-e2e:
    if: ...

              repo,
              context: process.env.CHECK_NAME
```

Reply:
> Can you converge both workflows with two different jobs? One for each event?

sounds good! thank you for the tip

Reply:
putting both jobs in one file caused to show this "Skipped" status check.

![image](https://github.com/elastic/kibana/assets/16325797/6bec929d-d0db-4adc-b568-1e01fbe5b1a9)

Reply:
Thanks for double checking this out. I certainly missed this particular behaviour :/